app/controllers/bathrooms_controller.rb

```diff
@@ -0,0 +1,62 @@
+class BathroomsController < ApplicationController
+  def index
+    @bathrooms = Bathroom.all
+  end
+
+  def new
+    @bathroom = Bathroom.new
+  end
+
+  def create
+    if logged_in?
+      @bathroom = Bathroom.new(bathroom_params)
+      @bathroom.user_id = session[:user_id]
+      if @bathroom.save
+        redirect_to root_path
+      else
+        render :new
+      end
+    else
+      redirect_to '/login'
+    end
+  end
+
+  def show
+    @bathroom = Bathroom.find(params[:id])
+  end
+
+  def edit
+    @bathroom = Bathroom.find(params[:id])
+    unless @bathroom.user.id == session[:user_id]
+      redirect_to "/login"
+    end
+  end
+
+  def update
+    @bathroom = Bathroom.find(params[:id])
+    if logged_in? && @bathroom.user.id == current_user.id
```

Comment:
Think about putting owned_by? as a method on bathroom. So you'd have 
`if @bathroom.owned_by?(current_user)`